staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1105,9 +1105,12 @@ type ManagedFieldsEntry struct {
 	// Time is timestamp of when these fields were set. It should always be empty if Operation is 'Apply'
 	// +optional
 	Time *Time `json:"time,omitempty" protobuf:"bytes,4,opt,name=time"`
-	// Fields identifies a set of fields.
+	// FieldsType is the discriminator for the different fields format and version.
+	// There is currently only one possible value: "FieldsJSONv1"
+	FieldsType string `json:"fieldsType,omitempty" protobuf:"bytes,6,opt,name=fieldsType"`
+	// FieldsJSONv1 holds the first JSON version format as described in the "Fields" type.
 	// +optional
-	Fields *Fields `json:"fields,omitempty" protobuf:"bytes,5,opt,name=fields,casttype=Fields"`
+	FieldsJSONv1 *FieldsJSONv1 `json:"fieldsJSONv1,omitempty" protobuf:"bytes,5,opt,name=fieldsJSONv1,casttype=FieldsJSONv1"`
```

Comment:
You need to tombstone the original `fields` value and at first glance, the `FieldsJSONv1` doesn't look compatible with the previously shipped protobuf value from 1.15 here: https://github.com/kubernetes/kubernetes/blob/release-1.15/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1133.

@liggitt is there some proto-magic that makes delta from 1.15 compatible that I'm not seeing?